atomic_reactor/dirs.py

```diff
@@ -256,29 +255,75 @@ def for_all_platforms_copy(self, action: FileCreationFunc) -> List[Path]:
                 file_path = build_dir / file_path
             file_path = file_path.resolve()
             if file_path == build_dir:
-                raise ValueError(
-                    f"{file_path} should not be added as a created directory."
-                )
+                raise ValueError(f"{file_path} should not be added as a created directory.")
             try:
                 file_path.relative_to(build_dir)
             except ValueError as e:
                 raise ValueError(
-                    f"File must be created inside the build directory. "
+                    f"File must be created inside the platform directory. "
                     f"File {file_path} is not allowed."
                 ) from e
             if not file_path.exists():
                 raise FileNotFoundError(
-                    f"{file_path} does not exist inside build directory."
+                    f"{file_path} does not exist inside the platform directory."
                 )
             the_new_files.append(file_path)
 
         for platform in self.platforms[1:]:
             for src_file in the_new_files:
-                dest = self.path / platform / src_file.relative_to(build_dir)
+                dest = self._root_path / platform / src_file.relative_to(build_dir)
                 if src_file.is_dir():
                     copytree(src_file, dest)
                 else:
                     dest.parent.mkdir(parents=True, exist_ok=True)
                     copyfile(src_file, dest)
 
         return the_new_files
+
+
+class ContextDir(BaseDataDir):
+    """Representing the context directory for the whole container build pipeline."""
+
+    @property
+    def workflow_data_file(self):
+        return self._root_path / "workflow.json"
+
+
+class RootBuildDir(BaseDataDir):
+    """A directory containing all artifacts for building images."""
```

Comment:
I still don't see that much value in making `ContextDir` behave the same way as `RootBuildDir`. Will the workflow data management involve the platform-specific behavior? I feel like the workflow data management that we need *currently* will only really touch workflow.json - the per-platform behavior will not be utilized at all until we get to the build-result.json - and even then, each build task only handles one platform, not multiple.